Orcasa-Platform/orcasa
# New app routing

This PR changes the app's routing so that each module can have specific routes.

Thanks to this, the details view of the Network module now lives on a separate route (`/network/organization/{ID}` or `/network/project/{ID}`. When navigating between the list and details view, the scroll position is also handled via dedicated hooks:
- On the details view, the page is always scrolled to the top
- On the lists view, the previous position is restored

The PR also replaces Recoil by useQueryState and Jotai since Recoil doesn't work well with Next.js 13's search params querying (`useSearchParams`) and scroll handling.

Finally, this PR also correctly handles the case where the user to share a URL where the basemap is not the default one. Previously, the default basemap would still be displayed, despite another being referenced in the URL.

## How to test

There are many cases to test:

- All modules
  - Moving the map and reloading to make sure its position is restored
  - Setting a non-default basemap on a module and reloading the page to make sure it is still the one shown
- Network module
  - Moving the map, navigating to a details view and reloading the page to make sure the map's position is restored
  - Navigating to a details view and reloading to check that the same details view is shown
  - Navigating to a details view to make sure that the sidebar is not scrolled down
  - Scrolling the list view, navigating to a details view and going back via the back link to check if the scroll's position in the list is restored

## Tracking

[ORC-224](https://vizzuality.atlassian.net/browse/ORC-224)

[ORC-224]: https://vizzuality.atlassian.net/browse/ORC-224?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
